src/atdf/field.rs

```diff
@@ -6,10 +6,15 @@ use crate::ElementExt;
 pub fn parse(
     bitfield_el: &xmltree::Element,
     value_groups: &atdf::values::ValueGroups,
+    mode_name: Option<String>,
 ) -> crate::Result<chip::Field> {
     debug_assert!(bitfield_el.name == "bitfield");
-
-    let name = bitfield_el.attr("name")?.clone();
+    let name;
+    if let Some(mode) = mode_name {
+        name = format!("{}_{}", mode, bitfield_el.attr("name")?.clone());
+    } else {
+        name = bitfield_el.attr("name")?.clone();
+    }
```

Comment:
In rust, you'd usually write this a tad differently:

```suggestion
    let name = if let Some(mode) = mode_name {
        format!("{}_{}", mode, bitfield_el.attr("name")?.clone())
    } else {
        bitfield_el.attr("name")?.clone()
    };
```